atlp-rwanda/e-comm-team-brogrammers-bn
# #184581353  User should be able to search a product

#### What does this PR do?
* This PR should add functionality to enable users to search for products by various parameters such as name, price range, category, and combinations of these parameters using an AND operation.
#### Description of Task to be completed?
* The task is to implement a search functionality that allows users to search for products based on various criteria such as name, price range, category, and their combinations using an AND operation. The implementation should be done using JavaScript, Node.js, Express, and PostgreSQL database.
#### How should this be manually tested?
* Launch the application
* Use postman or any other tool to test 
* Enter the search criteria such as name, price range, and category
* use GET method
Verify that the search results returned match the search criteria entered
Test different combinations of search criteria to verify the search functionality is working as expected.

#### Any background context you want to provide?
* To implement the search functionality, you can use SQL queries to search the PostgreSQL database. You can also use libraries such as Sequelize to make database queries easier. Additionally, you should consider adding indexes to the relevant database columns to improve search performance.
#### What are the relevant pivotal tracker/Trello stories?
*  [#184581353](https://www.pivotaltracker.com/n/projects/2632830/stories/184581353)

#### Screenshots (if appropriate)

![Screenshot from 2023-03-27 00-40-34](https://user-images.githubusercontent.com/101427855/227870999-a11830ee-5466-43af-bd6c-764cbbbc9223.png)

![Screenshot from 2023-03-27 00-41-14](https://user-images.githubusercontent.com/101427855/227871185-a2d7c24f-075d-492f-a781-a03430fc85bf.png)

![Screenshot from 2023-03-27 00-47-40](https://user-images.githubusercontent.com/101427855/227871235-42fd80e8-9564-4260-b9e7-5d18849d68ff.png)

![Screenshot from 2023-03-27 00-48-02](https://user-images.githubusercontent.com/101427855/227871323-8fa31473-2588-46fa-ae8a-08543d6266ba.png)

![Screenshot from 2023-03-27 11-32-41](https://user-images.githubusercontent.com/101427855/227887081-f51aad0e-b73b-4f8c-a3c2-454799bc9793.png)

#### Questions:
* N/A
